zarr/tests/test_convenience.py

```diff
@@ -202,23 +202,30 @@ def test_tree(zarr_version):
     assert str(zarr.tree(g1)) == str(g1.tree())
 
 
-# TODO: consolidated metadata currently only supported for v2
-
 @pytest.mark.parametrize('zarr_version', [2, 3])
 @pytest.mark.parametrize('with_chunk_store', [False, True], ids=['default', 'with_chunk_store'])
-def test_consolidate_metadata(with_chunk_store, zarr_version):
-
-    if zarr_version == 2:
-        MemoryStoreClass = MemoryStore
-        path = ''
-    else:
-        MemoryStoreClass = MemoryStoreV3
-        path = 'dataset'
-
+@pytest.mark.parametrize('stores_from_path', [False, True])
+def test_consolidate_metadata(with_chunk_store, zarr_version, stores_from_path):
```

Comment:
Outwith this PR, but this reminds me if v3 would make standardizing consolidated metadata easier. See https://github.com/zarr-developers/zarr-specs/issues/113

Reply:
also /zarr-developers/zarr-specs#136 related to where this metadata would be stored